Modules/IO/TIFF/src/itkTIFFImageIO.cxx

```diff
@@ -456,21 +456,33 @@ TIFFImageIO::ReadImageInformation()
       this->SetPixelType(RGBA);
   }
 
+  bool isPaletteShortType = false;
```

Comment:
I think this logic can be simpler and avoid the extra isPaletteShortType variable.

The logic before line 474 could be reverted.
m_ComponentType should not be changed after line 474.   (i.e. line 503 was in error before).

Reply:
You must be right. To be honest, I wrote this code 2-3 years ago. I'm going to run some test on a simplified version, and amend the pull request!